src/main/java/org/springframework/integration/dsl/config/IntegrationFlowBeanPostProcessor.java

```diff
@@ -143,6 +160,7 @@ private Object processStandardIntegrationFlow(StandardIntegrationFlow flow, Stri
 					endpointBeanName = generateBeanName(endpoint);
 				}
 				registerComponent(endpoint, endpointBeanName, beanName, registerSingleton);
+				integrationComponents.set(i, endpoint);
```

Comment:
This hack doesn't relate to the `Lifecycle`. For that purpose we really rely on that logic in the ctor to filter `Lifecycle` to additional property.
I need this replacement for artificial beans: `MessageChannelReference`, `FixedSubscriberChannelPrototype` etc.
All other `Spec` instances are here just coincidentally.
Although I agree with your that with this replacement fix we can revise the logic in the `StandardIntegrationFlow` for `Lifecycle`s.